### PR TITLE
Metadata network modifications

### PR DESCRIPTION
**Pull request by souissimai**

### CODE REVIEW

File: src/components/graph/menus/modification-list-item.js [hunk 1]
```diff
@@ -103,22 +110,52 @@ export const ModificationListItem = ({
...
-        });
+        const modificationValues = JSON.parse(modif.messageValues);
+
+        switch (modif.messageType) {
+            case MODIFICATION_TYPES.BRANCH_STATUS_MODIFICATION.type:
+                setComputedValues(
+                    getBranchStatusModificationValues(modificationValues)
+                );
+                break;
+            case MODIFICATION_TYPES.EQUIPMENT_ATTRIBUTE_MODIFICATION.type:
```
Comment (Meklo) [inline src/components/graph/menus/modification-list-item.js:141]: If i'm not mistaken this case wasn't handled before, what does it fixes or add ?

Comment (souissimai) [reply src/components/graph/menus/modification-list-item.js:141]: it's specific case that we should adapt after modification done on the back end

File: src/components/graph/menus/modification-list-item.js [hunk 2]
```diff
@@ -76,31 +77,37 @@ export const ModificationListItem = ({
...
-            case 'LINES_ATTACH_TO_SPLIT_LINES':
-                return modif.attachedLineId;
-            case 'DELETE_VOLTAGE_LEVEL_ON_LINE':
-                return modif.lineToAttachTo1Id + '/' + modif.lineToAttachTo2Id;
-            case 'DELETE_ATTACHING_LINE':
+        const modificationMetadata = JSON.parse(modif?.messageValues);
+
+        switch (modif.messageType) {
+            case MODIFICATION_TYPES.LINE_SPLIT_WITH_VOLTAGE_LEVEL.type:
+                return modificationMetadata?.lineToSplitId;
```
Comment (SlimaneAmar) [inline src/components/graph/menus/modification-list-item.js:84]: Remove `nullish` operator

File: src/components/graph/menus/modification-list-item.js [hunk 3]
```diff
@@ -185,7 +222,7 @@ export const ModificationListItem = ({
                             isEditableModification(modif) && (
                                 <IconButton
                                     onClick={() =>
-                                        onEdit(modif.uuid, modif?.type)
+                                        onEdit(modif.uuid, modif?.messageType)
```
Comment (SlimaneAmar) [inline src/components/graph/menus/modification-list-item.js:225]: Remove this same above

File: src/components/graph/menus/modification-list-item.js [hunk 4]
```diff
@@ -109,22 +116,52 @@ export const ModificationListItem = ({
...
             return;
         }
-        setComputedValues({
-            energizedEnd: modif.energizedVoltageLevelId,
-            computedLabel: <strong>{getComputedLabel()}</strong>,
-        });
+        const modificationValues = JSON.parse(modif.messageValues);
+
+        switch (modif.messageType) {
+            case MODIFICATION_TYPES.BRANCH_STATUS_MODIFICATION.type:
```
Comment (SlimaneAmar) [inline src/components/graph/menus/modification-list-item.js:142]: Rename all `computedValues` by 'computedLabelValues'
Also the state and `setComputedValues`

File: src/services/study/network-modifications.js [hunk 5]
```diff
@@ -1503,7 +1503,8 @@ export function deleteEquipment(
 export function fetchNetworkModifications(
     studyUuid,
     nodeUuid,
-    stashedModifications
+    stashedModifications,
+    onlyMetadata
```
Comment (SlimaneAmar) [inline src/services/study/network-modifications.js:1507]: `onlyMetadata` always equals to `true`
Remove it (back also)

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/components/graph/menus/modification-list-item.js [hunk 6]
```diff
@@ -27,7 +28,7 @@ const isEditableModification = (modif) => {
     if (!modif) {
         return false;
     }
-    return !nonEditableModificationTypes.has(modif.type);
+    return !nonEditableModificationTypes.has(modif.messageType);
```
[the remark this answers is not in the document]
Comment (souissimai) [reply src/components/graph/menus/modification-list-item.js:31]: it should be messageType, we don't have this info when metadata=true

**Review by SlimaneAmar — APPROVED** [review]

**Review by Meklo — APPROVED** [review]

**Pull request merged**